Results_Engine/Query/Group.cs

```diff
@@ -67,6 +70,88 @@ public static partial class Query
         }
 
         /***************************************************/
+
+        [Description("Group results by all Scenario type properties of the object.")]
+        [Input("results", "Results to group.")]
+        [Output("groupedResults", "Results grouped as List of Lists.")]
+        public static List<List<T>> GroupByScenario<T>(this IEnumerable<T> results) where T : IResult
+        {
+            if (results == null || !results.Any())
+                return new List<List<T>>();
+
+            return GroupByMultipleSelectors(results, IdentifierGetters(results.First().GetType(), typeof(ScenarioIdentifierAttribute)));
+        }
+
+        /***************************************************/
+
+        [Description("Group results by Object identifiers of the object.")]
+        [Input("results", "Results to group.")]
+        [Output("groupedResults", "Results grouped as List of Lists.")]
+        public static List<List<T>> GroupByObjectIdentifiers<T>(this IEnumerable<T> results) where T : IResult
+        {
+            if (results == null || !results.Any())
+                return new List<List<T>>();
+
+            return GroupByMultipleSelectors(results, IdentifierGetters(results.First().GetType(), typeof(ObjectIdentifierAttribute)));
+        }
+
+        /***************************************************/
+
+        [Description("Group results by all identifiers of the object.")]
+        [Input("results", "Results to group.")]
+        [Output("groupedResults", "Results grouped as List of Lists.")]
+        public static List<List<T>> GroupByAllIdentifiers<T>(this IEnumerable<T> results) where T : IResult
+        {
+            if (results == null || !results.Any())
+                return new List<List<T>>();
+
+            return GroupByMultipleSelectors(results, IdentifierGetters(results.First().GetType(), typeof(IdentifierAttribute)));
+        }
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        [Description("Groups by multiple selectors by contiously grouping by each provided selector.")]
+        private static List<List<T>> GroupByMultipleSelectors<T>(this IEnumerable<T> results, IEnumerable<Func<object, object>> funcs)
```

Comment:
```suggestion
        private static List<List<T>> GroupByMultipleSelectors<T>(this IEnumerable<T> results, IEnumerable<Func<object, object>> selectors)
```

Reply:
Yep, happy with this. WIll make as a commit from VS as it requires change lower down as well